serverless/serverless
# Limited Lambda Permission for Events

## What did you implement:

Closes #2023 



## How did you implement it:
Added `SourceArn` property for each `Lambda::Permission` resource depending on the event. The logical IDs of the `Lambda::Permission` resources had to be changed to:

* SNS: HelloLambdaPermissionTopicSNS
* S3: HelloLambdaPermissionBucketS3
* APIG: HelloLambdaPermissionUsersCreatePostApiGateway

I've updated the docs with the new logical IDs

Notes:

* This doesn't apply to stream events because they use different permission model via IAM statements, not `Lambda::Permission`



## How can we verify it:


Using the following config:

```yml
service: myService

provider:
  name: aws
  runtime: nodejs4.3

functions:
  hello:
    handler: handler.hello
    events:
      - http:
          path: users/create
          method: get
      - s3: serverless-limit-permissions-bucket
      - schedule: rate(10 minutes)
      - sns: serverless-limit-permissions-topic
```
Deploy and make sure your function is triggered successfully based on the event source. You can also check the SourceArn in the CF template with `--noDeploy`


## Todos:



***Is this ready for review?:*** YES
